src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Crypto.cs

```diff
@@ -170,5 +171,117 @@ internal static byte[] GetDynamicBuffer<THandle>(NegativeSizeReadMethod<THandle>
+        private static unsafe partial int SetMemoryTracking(delegate* unmanaged<MemoryOperation, UIntPtr, UIntPtr, int, char*, int, void> trackingCallback);
+
+        [StructLayout(LayoutKind.Sequential)]
+        private unsafe struct MemoryEntry
+        {
+            public int Size;
+            public int Line;
+            public char* File;
+        }
+
+        private enum MemoryOperation
+        {
+            Malloc = 1,
+            Realloc = 2,
+            Free = 3,
+        }
+
+        private static readonly unsafe nuint Offset = (nuint)sizeof(MemoryEntry);
+        private static HashSet<UIntPtr>? _allocations;
+
+        [UnmanagedCallersOnly]
+        private static unsafe void MemoryTrackinCallback(MemoryOperation operation, UIntPtr ptr, UIntPtr oldPtr, int size, char* file, int line)
+        {
+            ref MemoryEntry entry = ref *(MemoryEntry*)ptr;
+
+            Debug.Assert(entry.File != null);
+            Debug.Assert(ptr != UIntPtr.Zero);
+
+            switch (operation)
+            {
+                case MemoryOperation.Malloc:
+                    Debug.Assert(size == entry.Size);
+                    lock (_allocations!)
+                    {
+                        _allocations!.Add(ptr);
+                    }
+                    break;
+                case MemoryOperation.Realloc:
+                    lock (_allocations!)
+                    {
+                        if ((IntPtr)oldPtr != IntPtr.Zero)
+                        {
+                            _allocations!.Remove(oldPtr);
+                        }
+                        _allocations!.Add(ptr);
+                    }
+                    break;
+                case MemoryOperation.Free:
+                    lock (_allocations!)
+                    {
+                        _allocations!.Remove(ptr);
+                    }
+                    break;
+            }
+        }
+
+        public static unsafe void EnableTracking()
+        {
+            _allocations ??= new HashSet<UIntPtr>();
+            _allocations!.Clear();
+            SetMemoryTracking(&MemoryTrackinCallback);
+        }
+
+        public static unsafe void DisableTracking()
+        {
+            SetMemoryTracking(null);
+            _allocations!.Clear();
+        }
+
+        public static unsafe Tuple<UIntPtr, int, string>[] GetIncrementalAllocations()
+        {
+            if (_allocations == null || _allocations.Count == 0)
+            {
+                return Array.Empty<Tuple<UIntPtr, int, string>>();
+            }
+
+            lock (_allocations!)
+            {
+                Tuple<UIntPtr, int, string>[] allocations = new Tuple<UIntPtr, int, string>[_allocations.Count];
+                int index = 0;
+                foreach (UIntPtr ptr in _allocations)
+                {
+                    Span<MemoryEntry> entry = new Span<MemoryEntry>((void*)ptr, 1);
```

Comment:
```suggestion
                    ref MemoryEntry entry = ref *(MemoryEntry*)ptr;
```
No need to use Span here.